samples/csharp/MicrosoftGraphSDK/Submission.cs

```diff
@@ -157,5 +188,135 @@ public class Submission
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <param name="submissionId">Student submission id</param>
+        /// <param name="outcomeId">Outcome id</param>
+        /// <param name="educationOutcome">Education outcome object</param>
+        /// <returns>EducationOutcome</returns>
+        public static async Task<EducationOutcome> PatchOutcomeAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId,
+            string submissionId,
+            string outcomeId,
+            EducationPointsOutcome educationOutcome)
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Submissions[submissionId]
+                    .Outcomes[outcomeId]
+                    .PatchAsync(educationOutcome);
+            }
+            catch (Exception ex)
+            {
+                throw new GraphException($"PatchOutcomeAsync call: {ex.Message}", ex, classId, assignmentId, submissionId, educationOutcome);
+            }
+        }
+
+        /// <summary>
+        /// List all the outcomes associated with a submission
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <param name="submissionId">Student submission id</param>
+        /// <returns>EducationOutcomeCollectionResponse</returns>
+        public static async Task<EducationOutcomeCollectionResponse> GetSubmissionOutcomesAsync(
+            GraphServiceClient client,
+            string classId,
+            string assignmentId,
+            string submissionId)
+        {
+            try
+            {
+                return await client.Education
+                    .Classes[classId]
+                    .Assignments[assignmentId]
+                    .Submissions[submissionId]
+                    .Outcomes
+                    .GetAsync();
+            }
+            catch(Exception ex)
+            {
+                throw new GraphException($"GetSubmissionOutcomesAsync call: {ex.Message}", ex, classId, assignmentId, submissionId);
+            }
+        }
+
+        /// <summary>
+        /// Creating Resources folder for Submission
+        /// </summary>
+        /// <param name="client">Microsoft Graph service client</param>
+        /// <param name="classId">User class id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
+        /// <param name="submissionId">Student submission id</param>
+        /// <returns>EducationOutcomeCollectionResponse</returns>
```

Comment:
EducationSubmission

Reply:
updated to return type in headers to "EducationSubmission"